src/fenicsxconcrete/bcs.py

```diff
@@ -0,0 +1,36 @@
+
+from pydantic import ConfigDict, RootModel
+from pydantic.dataclasses import dataclass
+from pydantic.types import conlist
+from typing import Callable, NewType
+import numpy as np
+
+Marker = NewType('Marker', int)
+
+@dataclass(config=dict(arbitrary_types_allowed=True))
+class DirichletBCDefinition:
+    """
+    Definition of a time- and position-dependent Dirichlet boundary condition.
```

Comment:
How would the Dirichlet boundary condition be defined in time? Just passing a np.ndarray or a function?

Reply:
That would need to be a function $f(x,t)\mapsto y,\quad  x\in\partial\Omega$, so in python a `Callable[[np.ndarray, float],np.ndarray]` . For simplicity, a numpy array would be interpreted as a constant BC at every point in time